test/utils/factories/users.factory.ts

```diff
@@ -1,28 +1,92 @@
-import { Factory } from "@mikro-orm/seeder";
+import { EntityManager } from "@mikro-orm/core";
 
 import { faker } from "@faker-js/faker";
 
-import { UserProfile } from "@/common/entities/user-profiles.entity";
+import { Student } from "@/common/entities/students.entity";
+import { Teacher } from "@/common/entities/teachers.entity";
 import { User } from "@/common/entities/users.entity";
+import { EEducationLevel, EMedium } from "@/common/enums/students.enums";
+import { EUserType } from "@/common/enums/users.enums";
 
-export class UserFactory extends Factory<User> {
-  model = User;
-
-  definition(): Partial<User> {
-    return {
+export class UserFactory {
```

Comment:
The factory should be used to define the user interface and then create objects.
The db creation should be handled by helpers residing in `test/utils/helpers` and named `create-user-in-db.helpers.ts`